tools/abigen/internal/generate.go

```
@@ -267,3 +301,221 @@
+		// Try to acquire the lock again
+		lock, err = os.OpenFile(lockFile, os.O_CREATE|os.O_WRONLY|os.O_EXCL, 0600)
+		if err != nil {
+			return "", fmt.Errorf("could not acquire lock for downloading solc: %w", err)
+		}
+	} else if err != nil {
+		return "", fmt.Errorf("could not create lock file: %w", err)
+	}
+	
+	// Make sure to clean up the lock file when we're done
+	defer func() {
+		lock.Close()
```

golangci-lint (1.62.2)

Error return value of `lock.Close` is not checked

(errcheck)